### ISSUE TITLE
Add method to clear a Sidre view

### ISSUE THREAD

**Issue by kennyweiss** [LLNL/axom#509]
When working with Sidre recently, I noticed that there are no public method to clear a `sidre::View` and return it to the empty, unallocated, unapplied state.

Since this can be helpful in some cases, we should add it.
Perhaps `sidre::View::clear()` ?

It is likely that this function can be implemented in terms of existing functionality provided by `private` functions, e.g. `sidre::View::unapply()`.
https://github.com/LLNL/axom/blob/2de3ce8c1c505dfe7ce6f833601e517534a29b57/src/axom/sidre/core/View.hpp#L1359-L1369

**Comment by nselliott** [comment]
Yes, this is a good idea.  All the tools are there, or are simple one-line calls (such as `m_shape.clear()`).  `setBufferViewToEmpty()` gets you most of the way there if it's a View with an attached Buffer, but you also need to clear the schema, the shape, and the attributes.
